### PR TITLE
Make integer animation tracks be continuous when created

### PR DESCRIPTION
**Pull request by YeldhamDev**

### CODE REVIEW

File: editor/animation_track_editor.cpp [hunk 1]
```diff
@@ -4240,7 +4240,8 @@ AnimationTrackEditor::TrackIndices AnimationTrackEditor::_confirm_insert(InsertD
 				PropertyInfo h = _find_hint_for_track(animation->get_track_count() - 1, np);
 				animation->remove_track(animation->get_track_count() - 1); // Hack.
 
-				if (h.type == Variant::FLOAT ||
+				if (h.type == Variant::INT ||
```
Comment (Calinou) [inline editor/animation_track_editor.cpp:4243]: Vector2i, Vector3i and Rect2i should probably be added here too, as they use integer components.

Comment (YeldhamDev) [reply editor/animation_track_editor.cpp:4243]: Done.

**Review by TokageItLab — COMMENTED** [review]
Also I think BASIS can be added.

**Review by TokageItLab — COMMENTED** [review]
And, those cubic interpolation is done by `Animation::_cubic_interpolate` in `animation.cpp`, so that may need to be implement it.

**Comment by reduz** [comment]
I am not sure myself, a lot of things you can animate like states, frames, etc. are also integers. It just seems kind of odd to make integer tracks continuous by default. I would be happy if you can explain in which situations do you need this.

**Comment by reduz** [comment]
At much, we could make it dependant on the hint I think.

**Comment by KoBeWi** [comment]
While I don't much get the reasoning either (although vaguely remember it being discussed), I tested it and it doesn't seem to break anything, e.g. `frame` animations are the same as before.

**Comment by reduz** [comment]
@KoBeWi I think what changes is that the value is sent all the time, and I am not entirely sure if this will work with every property or if every possible property checks that the value is different before doing what needs to be done. If it does check, it may add a hidden performance issue in many cases associated to a state change.

**Comment by YeldhamDev** [comment]
> I would be happy if you can explain in which situations do you need this.

The reason why I made this PR was because I was trying to animate the margins of a `Control` node, which are integer values. And believe me, it didn't immediately click to me that the reason everything was just directly snapping into place was because the track was set to discrete mode.

**Pull request closed without merging**